test/service/test_manifest_async.py

```diff
@@ -185,73 +188,94 @@ def test(self, mock_uuid, mock_helper):
-                                                     partition=dict(index=0,
-                                                                    is_last=False,
-                                                                    file_name=None,
-                                                                    multipart_upload_id=None,
-                                                                    part_etags=None,
-                                                                    page_index=None,
-                                                                    is_last_page=None,
-                                                                    search_after=None))
-                            )
-                            mock_helper.describe_execution.assert_not_called()
-                            mock_helper.reset_mock()
-                            mock_helper.describe_execution.return_value = {'status': 'RUNNING'}
-                            params = None
-                        elif i == 1:
-                            mock_helper.start_execution.assert_not_called()
-                            mock_helper.describe_execution.assert_called_once()
-                            mock_helper.reset_mock()
-                            # simulate absence of output due eventual consistency
-                            mock_helper.describe_execution.return_value = {'status': 'SUCCEEDED'}
-                        elif i == 2:
-                            mock_helper.start_execution.assert_not_called()
-                            mock_helper.describe_execution.assert_called_once()
-                            mock_helper.reset_mock()
-                            mock_helper.describe_execution.return_value = {
-                                'status': 'SUCCEEDED',
-                                'output': json.dumps(
-                                    {
-                                        'manifest': {
-                                            'location': manifest_url,
-                                            'was_cached': False,
-                                            'format_': format_.value,
-                                            'catalog': self.catalog,
-                                            'filters': filters,
-                                            'object_key': object_key
-                                        }
-                                    }
+                    partition = dict(index=0,
+                                     is_last=False,
+                                     file_name=None,
+                                     multipart_upload_id=None,
+                                     part_etags=None,
+                                     page_index=None,
+                                     is_last_page=None,
+                                     search_after=None)
+                    get_manifest_states = [ManifestPartition.from_json(partition), manifest]
```

Comment:
I don't think this is realistic. `partition` is the first partition. The service would never return the first partition unchanged, since there is always some sign of progress. I don't think you need to cover multiple partitions in this test.